packages/analytics-js-integrations/src/integrations/Mixpanel/util.js

```diff
@@ -241,21 +240,32 @@ const getConsolidatedPageCalls = config =>
     : true;
 
 /**
- * Generates a custom event name for a page calls.
+ * Generates a custom event name for a page or screen.
  *
  * @param {Object} message - The message object
  * @param {string} userDefinedEventTemplate - The user-defined event template to be used for generating the event name.
+ * @throws {ConfigurationError} If the event template is missing.
  * @returns {string} The generated custom event name.
+ * @example
+ * const userDefinedEventTemplate = "Viewed {{ category }} {{ name }} Page";
+ * const message = {name: 'Home', properties: {category: 'Index'}};
+ * output: "Viewed Index Home Page"
  */
 const generatePageCustomEventName = (message, userDefinedEventTemplate) => {
-  try {
-    const eventTemplate = Handlebars.compile(userDefinedEventTemplate);
-    return eventTemplate(message);
-  } catch (error) {
-    logger.error(`Error generating custom event name: ${error.message}`);
-    // Return a default or empty string as a fallback
-    return 'Loaded a Page';
+  let eventName = userDefinedEventTemplate
+    .replace('{{ category }}', message.properties?.category || '')
+    .trim();
+  eventName = eventName.replace('{{ name }}', message.name || '').trim();
+  // Remove any extra space between placeholders
+  eventName = eventName.replace(/\s{2,}/g, ' ');
```

Comment:
We must do this only if the placeholders are replaced with empty strings. Let's not mess up the user's event format.

Reply:
So, should we remove this step completely of removing extra space from template?

Reply:
We have to do it more intelligently. Otherwise, we could end up disrupting the template.

For a template like the one below (note extra spaces are already present),
`  Viewed {{ name }} {{ category }} page  some   keyword`


If `name` is unavailable, the template becomes,
`  Viewed  {{ category }} page  some   keyword`

the cleaned template should have looked like this,
`  Viewed {{ category }} page  some   keyword`


If `category` is unavailable, the template becomes,
`  Viewed  page  some   keyword`

the cleaned template should have looked like this,
`  Viewed page  some   keyword`


So, you can see that we merge the two whitespaces into 1 only if a single whitespace on either side surrounds the handlebars.

Reply:
Updated the regex.

Reply:
No, that change is not adequate. Please reach out to me so we can discuss this.